libs/s25main/GamePlayer.cpp

```diff
@@ -1172,20 +1173,25 @@ bool GamePlayer::IsAttackable(const unsigned char playerId) const
         return GetPactState(PactType::NonAgressionPact, playerId) != PactState::Accepted;
 }
 
-void GamePlayer::OrderTroops(nobMilitary* goal, unsigned count, bool ignoresettingsendweakfirst) const
+void GamePlayer::OrderTroops(nobMilitary* goal, std::array<unsigned, NUM_SOLDIER_RANKS> counts, unsigned max) const
```

Comment:
```suggestion
void GamePlayer::OrderTroops(nobMilitary* goal, const std::array<unsigned, NUM_SOLDIER_RANKS>& counts, unsigned max) const
```

Reply:
I can't make this const like this it would get modified. It's just a few bytes, it's fine to copy it I think.